src/main/groovy/com/deploygate/gradle/plugins/internal/http/ApiClient.java

```diff
@@ -1,250 +1,47 @@
 package com.deploygate.gradle.plugins.internal.http;
 
-import com.deploygate.gradle.plugins.Config;
-import com.deploygate.gradle.plugins.internal.agp.AndroidGradlePlugin;
-import com.deploygate.gradle.plugins.internal.annotation.DeployGateInternal;
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
-import com.google.gson.JsonIOException;
-import com.google.gson.JsonSyntaxException;
+import com.deploygate.gradle.plugins.tasks.inputs.Credentials;
 import org.apache.hc.client5.http.HttpResponseException;
-import org.apache.hc.client5.http.classic.HttpClient;
-import org.apache.hc.client5.http.classic.methods.HttpGet;
 import org.apache.hc.client5.http.classic.methods.HttpPost;
-import org.apache.hc.client5.http.config.RequestConfig;
-import org.apache.hc.client5.http.impl.classic.AbstractHttpClientResponseHandler;
-import org.apache.hc.client5.http.impl.classic.HttpClientBuilder;
-import org.apache.hc.core5.http.ClassicHttpResponse;
-import org.apache.hc.core5.http.HttpEntity;
-import org.apache.hc.core5.http.message.BasicHeader;
-import org.apache.hc.core5.net.URIBuilder;
+import org.apache.hc.client5.http.classic.methods.HttpUriRequestBase;
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
-import org.jetbrains.annotations.VisibleForTesting;
-
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
 
 public class ApiClient {
```

Comment:
Extracted HttpClient stuff to new HttpClient class. ApiClient is like a retrofit's api service.